### PR TITLE
Communicate with Portal api to create events

### LINKED ISSUE #796: Communication with Portal Api
## Description
As a User, I want to create the events from the API in VMS.

### Update [Required]
- [ ] Fetch meet-ups from Portal 
- [ ]  Send them to createview

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
2 hours

### PR DESCRIPTION
**Pull request by anjali-dhanuka**
# Description
Create events from meet up data supplied by Portal

Fixes #796 

# Type of Change
- Code
- Quality Assurance
- User Interface
- Outreach
- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

# How Has This Been Tested?
image
image

# Checklist
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged 

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [x] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by anjali-dhanuka** [comment]
Hi @abhishekarya286 @ayushgarg1804 @kamsuri Please review

File: vms/vms/views.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+             start_date = meetup['start_date']
+             venue = meetup['venue']
+             name = meetup['event_name']
+             end_date = start_date  
+             event_details=[name, start_date, end_date]
+             event = create_event_with_details(event_details)
+             event.venue = venue
+             event.save()        
+        return HttpResponseRedirect(reverse('event:list'))
+
```
Comment (abhishekarya286) [inline vms/vms/views.py]: Extra lines

**Comment by anjali-dhanuka** [comment]
@abhishekarya286 That was not decided so I wanted to ask this, thus I just created a url. How should we proceed with fetching?

**Comment by abhishekarya286** [comment]
Please look at this for fetching data https://stackoverflow.com/a/19075476/7120232. @kamsuri any suggestions?

File: vms/vms/views.py [hunk 2]
```diff
@@ -3,10 +3,12 @@
...
 from django.http import HttpResponseRedirect
 from django.core.urlresolvers import reverse
-
+from datetime import datetime
 
 def get_meetup(request):
-    response = requests.get('http://127.0.0.1:8080/meetup/api/v1/request_meetup_data/')
+    date = str(datetime.today.date())
+    data = {'date': date}
+    response = requests.post('http://127.0.0.1:8080/meetup/api/v1/request_meetup_data/', data)
```
[the remark this answers is not in the document]
Comment (anjali-dhanuka) [reply vms/vms/views.py]: @may said that we are not interested in past events so only future events yes

File: vms/vms/urls.py [hunk 3]
```diff
@@ -2,7 +2,7 @@
 from django.conf.urls.i18n import i18n_patterns
 from django.conf.urls import include, url
 from django.contrib import admin
-
+from vms import views
```
Comment (Monal5031) [inline vms/vms/urls.py]: if you need only `get_meetup` so just import that.

File: vms/vms/views.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import requests
```
Comment (Monal5031) [inline vms/vms/views.py]: Sort the imports and modify them in sections.

File: vms/vms/views.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import json
+from shift.utils import create_event_with_details
+from django.http import HttpResponseRedirect
+from django.core.urlresolvers import reverse
+from datetime import datetime
+
+def get_meetup(request):
+    date = str(datetime.today.date())
+    data = {'date': date}
+    response = requests.post('http://127.0.0.1:8080/meetup/api/v1/request_meetup_data/', data)
```
Comment (Monal5031) [inline vms/vms/views.py]: the server IP should be dynamic IMO might change with deployment and test server too.

File: vms/vms/views.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    if response.status_code == 200:
+        json_data = json.loads(response.text)
+        for meetup in json_data:
+             start_date = meetup['start_date']
+             venue = meetup['venue']
+             name = meetup['event_name']
+             end_date = start_date
+             event_details=[name, start_date, end_date]
```
Comment (Monal5031) [inline vms/vms/views.py]: space around binary operator.

File: vms/vms/views.py (hunk 5 above)
Comment (anjali-dhanuka) [reply vms/vms/views.py]: Currently, both are not deployed so that depends upon the port which user enters, this will be changed while deployment.

Comment (Monal5031) [reply vms/vms/views.py]: this method is used in test too right? during test this url will change every time we run the tests its not localhost. We need to keep something dynamic here to cater the prod, dev and test at the same time.

Comment (Monal5031) [reply vms/vms/views.py]: https://docs.djangoproject.com/en/dev/ref/contrib/sites/#hooking-into-the-current-site-from-views

Take a look at this once.

**Comment by anjali-dhanuka** [comment]
@abhishekarya286 @kamsuri @Monal5031 This PR is not complete, @abhi20161997 has to send id of the meetups otherwise it's impossible to track which meetups have been converted to events.

**Comment by abhi20161997** [comment]
@anjali-dhanuka  It's not a very good practice to have multiple commits in a single PR. I guess you should squash the 4 commits into one, would be easier for people to review.

**Comment by abhishekarya286** [comment]
Squash your commits.

**Comment by abhishekarya286** [comment]
Fix the travis errors.

**Comment by abhishekarya286** [comment]
Squash the commits

**Comment by anjali-dhanuka** [comment]
@abhishekarya286 Done with squashing. please review.

**Pull request merged**